components/portal/messages/filters.js

```diff
@@ -67,6 +67,40 @@ define(['angular'], function(angular) {
           }
         });
       };
+    })
+     .filter('filterMessagesWithInvalidDates', function() {
+      var retVal = [];
```

Comment:
Let's move the `retVal` a line down into the returned function, rather than using it as a closure.  Right now if you were to do `var foo = $filter('filterMessagesWithInvalidDates')` and then execute foo twice, the second time you'd get the array with double the entries.

Stylistically we never do that, but why risk it.